roxie/ccd/ccddali.cpp

```diff
@@ -436,21 +436,38 @@ class CRoxieDaliHelper : implements IRoxieDaliHelper, public CInterface
     IMPLEMENT_IINTERFACE;
     CRoxieDaliHelper() : serverStatus(NULL), connectWatcher(this)
     {
-        userdesc.setown(createUserDescriptor());
-        const char *roxieUser = NULL;
-        const char *roxiePassword = NULL;
-        if (topology)
-        {
-            roxieUser = topology->queryProp("@ldapUser");
-            roxiePassword = topology->queryProp("@ldapPassword");
-        }
-        if (!roxieUser)
-            roxieUser = "roxie";
-        if (!roxiePassword)
-            roxiePassword = "";
-        StringBuffer password;
-        decrypt(password, roxiePassword);
-        userdesc->set(roxieUser, password.str());
+        if (oneShotRoxie && topology && topology->hasProp("@workunit"))
+        {
+            connect(ROXIE_DALI_CONNECT_TIMEOUT);
```

Comment:
looking at the implementation of connect() CRoxieDaliHelper::isConnected should be atomic.